include/seqan3/io/sequence/sequence_file_in.hpp

```diff
@@ -1,175 +1,911 @@
+        stream{std::move(_stream)}, format{file_format{}}
+    {
+        static_assert(meta::in<valid_formats, file_format>::value,
+                      "You selected a format that is not in the valid_formats of this file.");
+
+        // buffer first record
+        read_next_record();
+    }
+    //!\}
+
+    /*!\name Range interface
+     * \brief Provides functions for record based reading of the file.
+     * \{
+     */
+    /*!\brief Returns an iterator to current position in the file.
+     * \returns An iterator pointing to the current position in the file.
+     *
+     * Equals end() if the file is at end.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    iterator begin() noexcept
+    {
+        return {*this};
+    }
+
+    /*!\brief Returns a sentinel for comparison with iterator.
+     * \returns Iterator to the first element.
+     *
+     * This element acts as a placeholder; attempting to dereference it results in undefined behaviour.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    sentinel end() noexcept
+    {
+        return {};
+    }
+
+    //!\copydoc back
+    reference front() noexcept
+    {
+        return back();
+    }
+
+    /*!\brief Return the record we are currently at in the file.
+     * \returns A reference to the currently buffered record.
+     *
+     * This function returns a reference to the currently buffered record, it is identical to dereferencing begin()
+     * since begin also always points to the current record on single pass input ranges:
+     * ```cpp
+     * sequence_file_in fin{"/tmp/my.fasta"};
+     * auto it = begin(fin);
+     *
+     * // the following are equivalent:
+     * auto & rec0 = *it;
+     * auto & rec1 = fin.back();
+     * auto & rec2 = fin.front();
+     *
+     * // all become invalid after incrementing "it"!
+     * ```
+     *
+     * It most situations using the iterator interface or a range-based for-loop are preferable to using front() or
+     * back(), because you can only move to the next record via the iterator.
+     *
+     * In any case, don't forget the reference! If you want to save the data from the record elsewhere, use move:
+     *
+     * ```cpp
+     * sequence_file_in fin{"/tmp/my.fasta"};
+     *
+     * auto rec0 = std::move(fin.back());
+     * ```
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    reference back() noexcept
+    {
+        return record_buffer;
+    }
+    //!\}
+
+    /*!\name Tuple interface
+     * \brief Provides functions for field-based ("column"-based) reading.
+     * \{
+     */
+    //!\brief Read the entire file into internal buffers and retrieve the specified column.
+    template <field f>
+    friend auto & get(sequence_file_in & file)
```

Comment:
What about get(sequence_file_in && file)
Like in an interface as:
```cpp
auto all_seqs = get<field::SEQ>(sequence_file_in{"my.fasta"});
```

Would that be possible in general?

Reply:
it could be if I re-add the overloads for `&&`.

Reply:
done

Reply:
👍